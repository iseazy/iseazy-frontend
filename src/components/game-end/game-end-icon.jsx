export const GameEndIcon = () => (
    <svg xmlns="http://www.w3.org/2000/svg" width={32} height={32} >
        <defs>
        <style>
            {
            ".game-end__icon{fill:none;stroke:#999;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px}"
            }
        </style>
        </defs>
        <g transform="translate(-1 -1)">
        <circle
            className="game-end__icon"
            cx={15}
            cy={15}
            r={15}
            transform="translate(2 2)"
        />
        <path className="game-end__icon" d="M17 8v9l6 3" />
        </g>
    </svg>
)