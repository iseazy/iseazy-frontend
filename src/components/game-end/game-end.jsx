import { getFormattedTime } from '../../utils';
import { GameEndIcon } from './game-end-icon';

export const GameEnd = ({ time, onClick }) => (
    <div className="game-end">
            <div className="game-end__popup">
                <div className="game-end__title">
                    <div>Â¡Completado!</div>
                    <GameEndIcon />
                    <h1>{ getFormattedTime(time) }</h1>
                </div>
            <button className="game-end__button" onClick={ onClick}>Jugar otra vez</button>
        </div>
    </div>
);
