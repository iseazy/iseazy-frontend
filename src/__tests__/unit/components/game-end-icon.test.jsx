import { render } from '@testing-library/react';
import { GameEndIcon } from '../../../components/game-end/game-end-icon';

describe('<GameEndIcon />', () => {
    it('should render icon component', () => {
        const { asFragment } = render(<GameEndIcon />);

        expect(asFragment()).toMatchSnapshot();
    });
});
