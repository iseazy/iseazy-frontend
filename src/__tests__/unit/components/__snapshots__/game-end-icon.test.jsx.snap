// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameEndIcon /> should render icon component 1`] = `
<DocumentFragment>
  <svg
    height="32"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <style>
        .game-end__icon{fill:none;stroke:#999;stroke-linecap:round;stroke-linejoin:round;stroke-width:2px}
      </style>
    </defs>
    <g
      transform="translate(-1 -1)"
    >
      <circle
        class="game-end__icon"
        cx="15"
        cy="15"
        r="15"
        transform="translate(2 2)"
      />
      <path
        class="game-end__icon"
        d="M17 8v9l6 3"
      />
    </g>
  </svg>
</DocumentFragment>
`;
